<launch>
	<arg name="imu_node" value="false"/>
	<arg name="face_detect" value="true"/>
	<arg name="laser_scan" value="false"/>
	<arg name="android_joystick" value="false"/>
	<arg name="gmapping" value="false"/>
	<arg name="amcl" value="false"/>

	<node name="twistToMotors" pkg="chefbot_bringup" type="twist_to_motors" output="screen">
    	<!--remap from="cmd_vel" to="cmd_navigation"/-->
  	</node>

	<node name="base_controller_node" pkg="chefbot_bringup" type="base_controller">
	    <param name="publish_tf" value="true" /> <!-- false for ekf -->
	    <param name="use_imu" value="false" /> <!-- false for only encoder -->
	    <param name="publish_rate" value="10.0" />
	    <param name="linear_scale_positive" value="0.86" />
	    <param name="linear_scale_negative" value="0.86" />
	    <param name="angular_scale_positive" value="2.1" />
	    <param name="angular_scale_negative" value="2.1" />
	    <param name="angular_scale_accel" value="0.0" />
	    <param name="alpha" value="0.5" />
	  <!--
	    if (dth > 0) dth *= angular_scale_positive;
	  if (dth < 0) dth *= angular_scale_negative;
	  if (dxy_ave > 0) dxy_ave *= linear_scale_positive;
	  if (dxy_ave < 0) dxy_ave *= linear_scale_negative;
	-->
	</node>

	<!-- joystick -->
	<group if="$(eval android_joystick == true)">
		<node pkg="chefbot_joystick" type="joystick" name="joystick_node" />
	</group>

<!--
	<group if="$(eval imu_node == true)">
		<node pkg="tf" type="static_transform_publisher" name="baselink_imu_tf_publisher" args="0.15 0 0 0 0 0 1  /base_link imu  100"/>

		<include file="$(find imu_complementary_filter)/launch/complementary_filter.launch"/>

		 <node name="base_controller_imu_node" pkg="chefbot_bringup" type="base_controller_imu">
				<param name="publish_tf" value="true" />
				<param name="use_imu" value="false" />
	      <param name="publish_rate" value="10.0" />
	      <param name="linear_scale_positive" value="0.86" />
	      <param name="linear_scale_negative" value="0.86" />
	      <param name="angular_scale_positive" value="2.1" />
	      <param name="angular_scale_negative" value="2.1" />
	      <param name="angular_scale_accel" value="0.0" />
	      <param name="alpha" value="0.5" />
	  </node>
	</group>
      -->
  	<group if="$(eval face_detect == true)">
   		<include file="$(find chefbot_facedetection)/launch/start_tracking_usbcam.launch" />
  	</group>

	<group if="$(eval laser_scan == true)">
	 	<include file="$(find rplidar_ros)/launch/rplidar.launch" />
	 	<node name="baselink_laser_tf" type="static_transform_publisher" pkg="tf" args="0 0 0.1 0 0 0 1  /base_link laser  100"/>
	</group>

	<group if="$(eval gmapping == true)">
	 	<include file="$(find chefbot_slam)/launch/gmapping_demo.launch" />
	</group>

	<group if="$(eval amcl == true)">
	 	<include file="$(find chefbot_slam)/launch/amcl_demo.launch" />
	</group>
</launch>
